### PR TITLE
API: Update Kevin Progress

### PR DESCRIPTION
**Pull request by maingockien01**
- Added routes for endpoints environment, snapshot
- Added swagger page for API docs at /docs
- Added database connection files (need to update dbUri for the actual database)
- Added models/interface for environments module map

### CODE REVIEW

File: API/src/config.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const dbConnection = {
+    database: '',
```
Comment (jeff-knurek) [inline API/src/config.ts:2]: this is a good placeholder, but these values should come from ENV vars, not a config file - so maybe you can already build that logic in

File: API/src/core/ApiError.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    ) {
+        super(type);
+    }
+
+    public handle(res: Response): Response {
+        return this.response.send(res);
+    }
+}
+
+export class BadTokenError extends ApiError {
```
Comment (jeff-knurek) [inline API/src/core/ApiError.ts:36]: same comment as in the other PR, we won't have auth on the api for a while, so having token error/validation will just be unused code

File: API/src/core/logger.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+export interface ILogger {
+    info(message: string);
+    warn(e: Error, message?: string);
+}
+
+class Logger implements ILogger {
```
Comment (jeff-knurek) [inline API/src/core/logger.ts:6]: :+1:

File: API/src/main.ts [hunk 4]
```diff
@@ -1,8 +1,24 @@
...
+
+    const options = new DocumentBuilder()
+        .setTitle('Project Lockdown API')
+        .setDescription('This the api of project lockdown')
+        .setVersion('2.0')
+        .addTag('ProjectLockdown')
+        .build();
+
+    const apiDocument = SwaggerModule.createDocument(app, options);
+    SwaggerModule.setup('/docs', app, apiDocument);
```
Comment (jeff-knurek) [inline API/src/main.ts:19]: I don't think we want to combine Swagger with Stoplight. I'm not a big fan of Stoplight, so we can switch, but not in this PR, and instead we'll need to start a discussion https://github.com/TheIOFoundation/ProjectLockdown/discussions

**Review by jeff-knurek — COMMENTED** [review]
sharing some thoughts from the first half of the PR, will chat directly in slack about the other thoughts

File: API/src/main.ts (hunk 4 above)
Comment (maingockien01) [reply API/src/main.ts:19]: okay, I'll do research about them a bit.

File: API/src/config.ts (hunk 1 above)
Comment (maingockien01) [reply API/src/config.ts:2]: I think I will add a function to read ENV var and create that object later.

Comment (maingockien01) [reply API/src/config.ts]: @jeff-knurek #399 I made the change for this in the PR

**Pull request closed without merging**